### DISCUSSION TITLE
Bind a script to a button?

### DISCUSSION THREAD

**Discussion by please-fill-out-this-field** [AntiMicroX/antimicrox#453] [Q&A]
Hello!

README says I can bind *scripts and executables* to controller buttons, but how? I can't find any appropriate menus - there are all kinds of options for binding a keyboard/mouse button but nothing about scripts. Typing stuff in the "Action" field has no effect at all.
Let's say I want to be able to execute `zenity --notification --text="Hello world"` by pressing the SELECT button of my DS3. How do I do that?

![image](https://user-images.githubusercontent.com/50292908/170327619-c49574e4-dd8f-41d8-aa5d-f694258a35c5.png)

Thanks.

antimicrox 3.2.3

**Comment by pktiuk** [comment]
Press `Advanced` Button

**Reply by please-fill-out-this-field** [reply]
> Press `Advanced` Button

![image](https://user-images.githubusercontent.com/50292908/170344826-75701902-f58a-4d98-a8de-4d86fc85e2aa.png)

But I don't want it to trigger any keypress...

**Comment by please-fill-out-this-field** [comment]
Okay I managed to configure it.
The process was pretty weird and unintuitive though -- I had to assign some button before I could open the `Advanced` window just to discard that button a few seconds later. I would have never figured it out myself. It would probably be better to allow opening `Advanced` for unset (`[NO KEY]`) buttons.

**Reply by frankhuurman** [reply]
Ok wow I already gave up after trying for about 30 minutes and uninstalled it but seeing your comment made me try one more time and dang this is really unintuitive.

For anyone encountering this in the future:
- Click the button you want to set a script/executable to run
- Select a random key from the keyboard first, else the Advanced section won't open. Click advanced after this
- With the assignments tab on the left selected, you'll see your random keyboard selection on the right. Click it and press "delete" from the options on the right.
- Select the only empty field that was next to your random keyboard selection before(be careful not to click the little square itself but just outside of it else you're setting a new keybind again)
- In the dropdown that says Blank or KB/M select the Execute option
- In the new fields below, type the path to your binary/executable or browse to it with the button behind it. For testing purposes you could type /bin/gnome-terminal there if you're running on a Gnome environment
- Don't forget to press "insert" because pressing "close" will not save the stuff you just entered
- Now you can press close and it should launch your terminal once you press that button now
